CONTRIBUTING.md

```diff
@@ -0,0 +1,92 @@
+[GitHub flow]: https://docs.github.com/en/get-started/quickstart/github-flow
+[Collaborating with pull requests]: https://docs.github.com/en/github/collaborating-with-pull-requests
+
+## Getting Started
+
+### Issues
+
+#### Create a New Issue
+
+If you spot a problem with the docs, [search if an issue already exists]. If a related issue doesn't exist, you can open a new issue using the [issue form].
+
+[search if an issue already exists]: https://docs.github.com/en/github/searching-for-information-on-github/searching-on-github/searching-issues-and-pull-requests#search-by-the-title-body-or-comments
+[issue form]: https://github.com/esp-rs/esp-hal/issues/new/
+
+#### Solve an Issue
+
+Scan through our [existing issues] to find one that interests you. You can narrow down the search using labels as filters. If you find an issue to work on, you are welcome to open a PR with a fix.
+
+It's recommended that you comment in the relevant issue, mentioning that you are actively working on it, however this is not a requirement.
+
+[existing issues]: https://github.com/esp-rs/esp-hal/issues
+
+### Make Changes
+
+1. Fork the repository.
+   - Using GitHub Desktop:
+     - [Getting started with GitHub Desktop] will guide you through setting up Desktop.
+     - Once Desktop is set up, you can use it to [fork the repo!]
+   - Using the command line:
+     - [Fork the repo] so that you can make your changes without affecting the original project until you're ready to merge them.
+2. Install or update to the latest version of Rust. See [rustup.rs] for more information.
+3. Create a working branch and start with your changes!
+
+[Getting started with GitHub Desktop]: https://docs.github.com/en/desktop/installing-and-configuring-github-desktop/getting-started-with-github-desktop
+[fork the repo!]: https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/cloning-and-forking-repositories-from-github-desktop
+[Fork the repo]: https://docs.github.com/en/github/getting-started-with-github/fork-a-repo#fork-an-example-repository
+[rustup.rs]: https://rustup.rs/
+
+### Commit Your Update
+
+Commit the changes once you are happy with them. Don't forget to self-review to speed up the review process.
+
+We _strongly_ recommend that you use the supplied `pre-commit` Git hook, which will ensure that all source code has been formatted correctly prior to committing. See the [Git documentation] for more information on hooks.
+
+The `pre-commit` hook can be installed by running the following command in a terminal, from the root of the repository:
+
+```shell
+cp pre-commit .git/hooks/pre-commit
+```
+
+[Git documentation]: https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks
+
+### Pull Request
+
+When you're finished with the changes, create a pull request, also known as a PR.
+
+- Fill the "Ready for review" template so that we can review your PR. This template helps reviewers understand your changes as well as the purpose of your pull request.
```

Comment:
What's the "Ready for review" template? There is the PR template but that doesn't mention this phrase so this line might be somewhat confusing. Or, maybe I'm just overthinking.